enterprise/server/execution_search_service/execution_search_service.go

```diff
@@ -53,6 +59,44 @@ type ExecutionWithInvocationId struct {
 	invocationID string
 }
 
+func (s *ExecutionSearchService) FetchExecutionRequestMetadata(ctx context.Context, invocationID string, execIDs []string) (map[string]*repb.RequestMetadata, error) {
+	if !*enableFetchExecutionRequestMetadata {
+		return make(map[string]*repb.RequestMetadata), nil
+	}
+	if s.oh == nil {
+		return nil, status.UnavailableError("OLAP DB is required to search execution request metadata")
+	}
+	u, err := s.env.GetAuthenticator().AuthenticatedUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if u.GetGroupID() == "" {
+		return nil, status.InvalidArgumentError("failed to find user's group when searching executions request metadata.")
+	}
+	if err := authutil.AuthorizeGroupAccess(ctx, s.env, u.GetGroupID()); err != nil {
+		return nil, err
+	}
+	q := query_builder.NewQuery(`SELECT * FROM "Executions"`)
+	// Always filter to the currently selected (and authorized) group.
+	q.AddWhereClause("group_id = ?", u.GetGroupID())
+	q.AddWhereClause("invocation_uuid IN ?", invocationID)
```

Comment:
should this be `=` instead of `IN`? also do we need to replace all `"-"` in the ID with `""` like we do here: https://github.com/buildbuddy-io/buildbuddy/blob/7cf641453c36c0f5f2c81eb92dccaa64cc05645a/enterprise/server/util/execution/execution.go#L25